src/i2c.rs

```diff
@@ -0,0 +1,217 @@
+use std::marker::PhantomData;
+
+use crate::bindings::{
+    CgosI2CCount, CgosI2CGetFrequency, CgosI2CGetMaxFrequency, CgosI2CIsAvailable, CgosI2CRead,
+    CgosI2CReadRegister, CgosI2CSetFrequency, CgosI2CType, CgosI2CWrite, CgosI2CWriteReadCombined,
+    CgosI2CWriteRegister, CGOS_I2C_TYPE_DDC, CGOS_I2C_TYPE_PRIMARY, CGOS_I2C_TYPE_SMB,
+    CGOS_I2C_TYPE_UNKNOWN,
+};
+
+/// Error type for I2c operations
+#[derive(PartialEq, Eq, Clone, Debug)]
```

Comment:
This can also be `Copy`

```suggestion
#[derive(PartialEq, Eq, Copy, Clone, Debug)]
```